brave/brave-core
# Automatically redirect gateway resources

This introduces a new option that is off by default to use the x-ipfs-path header.
When it is present it will use the configured IPFS node.


Resolves https://github.com/brave/brave-browser/issues/11564

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.

## Test Plan:

- Load a resource like this: https://cloudflare-ipfs.com/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/ it should load it without redirecting.
- Turn the new option in preferences on to automatically redirect network settings
- Load a resource like this again: https://cloudflare-ipfs.com/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/ it should redirect to the configured gateway. 
- Repeat with a local node installed, it should go to the local node.
